Kernel/Arch/riscv64/Interrupts.cpp

```diff
@@ -11,9 +11,13 @@
 #include <Kernel/Arch/Interrupts.h>
 #include <Kernel/Arch/TrapFrame.h>
 #include <Kernel/Interrupts/GenericInterruptHandler.h>
+#include <Kernel/Interrupts/SharedIRQHandler.h>
+#include <Kernel/Interrupts/UnhandledInterruptHandler.h>
 
 namespace Kernel {
 
+static Array<GenericInterruptHandler*, 64> s_interrupt_handlers;
```

Comment:
Does this have to be a raw pointer, or can it be an OwnPtr? That would save the manual memory management in the register and unregister functions.

Reply:
The entire registering and unregistering code is copied from x86 and aarch64.
We should probably share that code somehow. Don't know if I should only fix this for RISC-V.

Reply:
another FIXME is in order

Reply:
are you fine with these FIXMEs?